analyzers/src/SonarAnalyzer.Common/Rules/RouteTemplateShouldNotStartWithSlashBase.cs

```diff
@@ -74,18 +74,25 @@ private void ReportIssues(SonarSymbolReportingContext context, INamedTypeSymbol
         var issueMessage = controllerSymbol.GetAttributes().Any(x => x.AttributeClass.IsAny(KnownType.RouteAttributes) || x.AttributeClass.Is(KnownType.System_Web_Mvc_RoutePrefixAttribute))
             ? MessageOnlyActions
             : MessageActionsAndController;
-        var attributeLocations = actions.SelectMany(x => x.RouteParameters).Select(x => x.Key);
 
-        if (ControllerDeclarationSyntax(controllerSymbol) is { } controllerSyntax
-            && Language.Syntax.NodeIdentifier(controllerSyntax) is { } identifier)
+        if (ControllerDeclarationSyntax(controllerSymbol) is { Length: > 0 } controllerSyntaxDeclarations)
         {
-            context.ReportIssue(Language.GeneratedCodeRecognizer, Diagnostic.Create(Rule, identifier.GetLocation(), attributeLocations, issueMessage));
+            var classLocations = controllerSyntaxDeclarations.Select(x => Language.Syntax.NodeIdentifier(x)?.GetLocation());
+            var secondaryLocations = FindActionSecondaryLocations(actions).Concat(classLocations.Skip(1));
+            context.ReportIssue(Language.GeneratedCodeRecognizer, Diagnostic.Create(Rule, classLocations.First(), secondaryLocations, issueMessage));
         }
 
-        SyntaxNode ControllerDeclarationSyntax(INamedTypeSymbol symbol) =>
+        SyntaxNode[] ControllerDeclarationSyntax(INamedTypeSymbol symbol) =>
             symbol.DeclaringSyntaxReferences
-            .FirstOrDefault(x => !x.GetSyntax().SyntaxTree.IsGenerated(Language.GeneratedCodeRecognizer))
-            ?.GetSyntax();
+            .Select(x => x.GetSyntax())
+            .Where(x => !x.SyntaxTree.IsGenerated(Language.GeneratedCodeRecognizer))
+            .ToArray();
+
+        IEnumerable<Location> FindActionSecondaryLocations(ConcurrentStack<ControllerActionInfo> actions) =>
+            actions
+                .Where(x => !x.Action.GetFirstSyntaxRef().SyntaxTree.IsGenerated(Language.GeneratedCodeRecognizer)) // We don't report on autogenerated actions
```

Comment:
@antoine-vigneau-sonarsource and @costin-zaharia-sonarsource 
What is your take on secondary locations in generated code?

I would argue that this is valid, as it might be the violation only becomes apparent if you take the auto-generated part into account as a developer. Otherwise, you might be puzzled as to why there is an issue because you do not see the full picture by only looking at your own code.